File: Multi_Language_CodeExamples/elixir/structs.exs
defmodule Person do
  defstruct name: "", age: 0
  
  def greet(%Person{name: name, age: age}) do
    "Hello, I'm #{name} and I'm #{age} years old"
  end
end

person = %Person{name: "Karen", age: 33}
IO.puts(Person.greet(person))

File: Multi_Language_CodeExamples/go/structs.go
package main

import "fmt"

type Person struct {
    Name string
    Age  int
}

func (p Person) Greet() {
    fmt.Printf("Hello, I'm %s and I'm %d years old\n", p.Name, p.Age)
}

func main() {
    person := Person{Name: "Charlie", Age: 35}
    person.Greet()
}